### ISSUE TITLE
Failed to build Open3D's SurfaceReconstructionPoisson.cpp

### ISSUE THREAD

**Issue by jiapei100** [mkazhdan/PoissonRecon#237]
@mkazhdan 

Hi, can you please take a look at this particular issue? `https://github.com/isl-org/Open3D/issues/5479` .
It looks it's a compatibility issue with current **PoissonRecon** ?

**Comment by mkazhdan** [comment]
Do you have a similar problem when using the original source code?

**Comment by jiapei100** [comment]
@mkazhdan 

1. Clearly, **Open3D** is using an old version of **[PoissonRecon](https://github.com/mkazhdan/PoissonRecon)**, please refer to [Open3D's possionrecon.cmake](https://github.com/isl-org/Open3D/blob/master/3rdparty/possionrecon/possionrecon.cmake).


2. And, I can see there are so many similarties, as well as differences between these 2 files:

[PoissonRecon.cpp](https://github.com/mkazhdan/PoissonRecon/blob/master/Src/PoissonRecon.cpp) **vs**. [SurfaceReconstructionPoisson.cpp](https://github.com/isl-org/Open3D/blob/master/cpp/open3d/geometry/SurfaceReconstructionPoisson.cpp)

Clearly. **Open3D** is trying to re-implement your [PoissonRecon.cpp](https://github.com/mkazhdan/PoissonRecon/blob/master/Src/PoissonRecon.cpp), in order to take their **Open3DData, Open3DPointStream, and Open3DVertex** etc...

**Comment by mkazhdan** [comment]
That's a tough one.
I believe I spend a certain amount of time updating the PoissonRecon so that the templated code would be cross-platform.
If the new code compiles on your machine but the Open3D code does not, it may make sense to talk to Open3D about updating their integration of the reconstruction code.

**Comment by jiapei100** [comment]
@mkazhdan 

Yeah, I tried that already. in [open3d - geometry - CMakeLists.txt](https://github.com/isl-org/Open3D/blob/master/cpp/open3d/geometry/CMakeLists.txt), I commented out **SurfaceReconstructionPoisson.cpp**. I can proceed until the final stage (every file compiles successfully, but the final link failed. )

For now, I'm trying to make current **Open3D** compatible with your **up-to-date PoissonRecon** ...

Hmmmm....

**Comment by cosama** [comment]
I just downloaded the version here and run it on some of my point clouds and it seems to do better than `open3d`. I thought I share my own experience looking into this, might help you to figure things out @jiapei100 

`open3d` uses version 12.0. I think their version is at commit hash `03f73754e994eb388de63285d3a2772493168e8a`, but they stripped out `.git` so it is hard to be sure. The only changes I can see is removing the code specific to `GCC < 5` in some of the `FEMTree*` files.

I would really love to see `open3d` updated to this version and maybe extended to include a few more parameters. @jiapei100 If you have something working please open a merge request over at `open3d` with some updated code.
